http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -364,27 +370,44 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
         s"googleIamDAO.addIamPolicyBindingOnServiceAccount for GSA ${gsa.value} & KSA ${ksaName.value}"
       ).compile.lastOrError
 
-      // helm install galaxy and wait
-      _ <- installGalaxy(
-        helmAuthContext,
-        app.appName,
-        app.release,
-        dbCluster,
-        dbApp.nodepool.nodepoolName,
-        namespaceName,
-        app.auditInfo.creator,
-        app.customEnvironmentVariables,
-        ksaName,
-        nfsDisk
-      )
+      // helm install and wait
+      _ <- app.appType match {
+        case AppType.Galaxy =>
+          installGalaxy(
+            helmAuthContext,
+            app.appName,
+            app.release,
+            dbCluster,
+            dbApp.nodepool.nodepoolName,
+            namespaceName,
+            app.auditInfo.creator,
+            app.customEnvironmentVariables,
+            ksaName,
+            nfsDisk
+          )
+        case AppType.Custom =>
+          installCustomApp(
+            app.id,
+            app.appName,
+            app.release,
+            dbCluster,
+            googleCluster,
+            dbApp.nodepool.nodepoolName,
+            namespaceName,
+            nfsDisk,
+            app.descriptorPath,
+            app.extraArgs,
+            app.customEnvironmentVariables
+          )
+      }
 
       _ <- logger.info(ctx.loggingCtx)(
         s"Finished app creation for app ${app.appName.value} in cluster ${gkeClusterId.toString}"
       )
 
       // TODO: this logic will be changed in the next PR
       isUsedByGalaxy <- persistentDiskQuery.isUsedByGalaxy(diskId).transaction
-      _ <- if (isUsedByGalaxy.exists(identity)) F.unit
+      _ <- if (isUsedByGalaxy.exists(identity) || app.appType != AppType.Galaxy) F.unit
```

Comment:
should this be `app.appType == AppType.Galaxy`?

Reply:
No, we want to not do any of the below logic for any apps besides galaxy (as its custom disk logic)

Reply:
nvm..I misread the condition the first time...